stats_test.go

```diff
@@ -0,0 +1,131 @@
+package codeowners
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestCalculateOwnershipStats(t *testing.T) {
+	tests := []struct {
+		name     string
+		files    Owners
+		expected OwnerStats
+	}{
+		{
+			name:  "empty files list",
+			files: Owners{},
+			expected: OwnerStats{
+				TotalFiles:    0,
+				OwnedFiled:    0,
+				UnownedFiles:  0,
+				TotalOwners:   0,
+				FilesPerOwner: []FilesPerOwner{},
+			},
+		},
+		{
+			name: "single owned file",
+			files: Owners{
+				{
+					Path:   "file1.txt",
+					Owners: []string{"@teamA"},
+				},
+			},
+			expected: OwnerStats{
+				TotalFiles:   1,
+				OwnedFiled:   1,
+				UnownedFiles: 0,
+				TotalOwners:  1,
+				FilesPerOwner: []FilesPerOwner{
+					{
+						Owner:      "@teamA",
+						Count:      1,
+						Percentage: 100.0,
+					},
+				},
+			},
+		},
+		{
+			name: "single unowned file",
+			files: Owners{
+				{
+					Path:   "file1.txt",
+					Owners: []string{"(unowned)"},
+				},
+			},
+			expected: OwnerStats{
+				TotalFiles:   1,
+				OwnedFiled:   0,
+				UnownedFiles: 1,
+				TotalOwners:  0,
+				FilesPerOwner: []FilesPerOwner{
+					{
+						Owner:      "(unowned)",
+						Count:      1,
+						Percentage: 100.0,
+					},
+				},
+			},
+		},
+		{
+			name: "multiple files with different owners",
+			files: Owners{
+				{
+					Path:   "file1.txt",
+					Owners: []string{"@teamA", "@teamB"},
+				},
+				{
+					Path:   "file2.txt",
+					Owners: []string{"@teamA"},
+				},
+				{
+					Path:   "file3.txt",
+					Owners: []string{"(unowned)"},
+				},
+			},
+			expected: OwnerStats{
+				TotalFiles:   3,
+				OwnedFiled:   2,
+				UnownedFiles: 1,
+				TotalOwners:  2,
+				FilesPerOwner: []FilesPerOwner{
+					{
+						Owner:      "@teamA",
+						Count:      2,
+						Percentage: 66.67,
+					},
+					{
+						Owner:      "@teamB",
+						Count:      1,
+						Percentage: 33.33,
+					},
+					{
+						Owner:      "(unowned)",
+						Count:      1,
+						Percentage: 33.33,
+					},
+				},
```

Comment:
One interesting thing to note about this implementation: `Percentage` will often add up to more than 100%, because more than one team can own a given file. I _think_ this is reasonable, but I'm open to other suggestions here.

Reply:
Nice, thanks for calling that out. I think it sounds reasonable too, but maybe we can be a little more clear in the helpdoc about what the semantics are. If that tells the user, "this command will answer you when you have the question <x>, " then i think we'll have nice ux and avoid any confusion here. (Other existing command names (hopefully) get some of that for free by virtue of their name.)

Reply:
Yeah, I can update the docs with this, good call.

Reply:
Okay, this is done now. Let me know what you think.

Reply:
Let's ship it!